### PR TITLE
The Supermatter hallucination effect no longer affects people with robotic eyes

### PR DESCRIPTION
**Pull request by Ziiro**
:cl:
add: The Supermatter hallucination effect no longer affects people with robotic eyes
/:cl:

Fixes https://github.com/OracleStation/OracleStation/issues/411

### CODE REVIEW

**Comment by Joseph-Case** [comment]
Add fixes #411 or whatever the issue was to the description

**Comment by SamHPurp** [comment]
Your changelog should be `fix` not `add`

**Comment by Ziiro** [comment]
Okay the code doesn't actually work, I was testing wrong. So nevermind on that. Back to the drawing board.

**Comment by Ziiro** [comment]
Well that tile is a mouthful, but it works now.

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -168,7 +168,8 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
 	for(var/mob/living/carbon/human/H in viewers(range, src))
 		if(H != user)
 			continue
-		if(!istype(H.glasses, /obj/item/clothing/glasses/meson))
+		var/obj/item/organ/eyes/eyes = H.getorganslot("eye_sight")
+		if(!istype(H.glasses, /obj/item/clothing/glasses/meson) && !istype(eyes, /obj/item/organ/eyes/robotic))
```
Comment (FlattestGuitar) [inline code/modules/power/supermatter/supermatter.dm]: do

```
if(istype(H.glasses, /obj/item/clothing/glasses/meson))
    continue
var/obj/item/organ/eyes/eyes = H.getorganslot("eye_sight")
if(eyes && eyes.status != ORGAN_ROBOTIC)
    continue```

Comment (Ziiro) [reply code/modules/power/supermatter/supermatter.dm]: I tried to make this block of code work, but I couldn't. Where does the `to_chat(H, "<span class='danger'>You get headaches just from looking at it.</span>")` go? Replace the second continue maybe?

**Comment by SamHPurp** [comment]
I feel like I'm missing something - why is this all under `examine()`

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -360,7 +364,8 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_shard)
 		air_update_turf()
 
 	for(var/mob/living/carbon/human/l in view(src, HALLUCINATION_RANGE(power))) // If they can see it without mesons on.  Bad on them.
-		if(!istype(l.glasses, /obj/item/clothing/glasses/meson))
+		var/obj/item/organ/eyes/eyes = l.getorganslot("eye_sight")
+		if(!istype(l.glasses, /obj/item/clothing/glasses/meson) && !istype(eyes, /obj/item/organ/eyes/robotic))
```
Comment (FlattestGuitar) [inline code/modules/power/supermatter/supermatter.dm]: Like with the previous check, this should really check for the `status` of the organ, not its path. The path doesn't guarantee anything.

**Comment by Ziiro** [comment]
@SamHPurp The examine text exists to print a warning if you're staring at it in a way that'll affect you. i.e: Without mesons or robotic eyes.

**Review by SamHPurp — APPROVED** [review]

**Pull request merged**